pallets/ah-migrator/src/lib.rs

```diff
@@ -717,11 +805,42 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(_: BlockNumberFor<T>) -> Weight {
-			Weight::zero()
+			T::AhWeightInfo::on_finalize()
+		}
+
+		fn on_finalize(_: BlockNumberFor<T>) {
+			let (processed, _) = XcmDataMessageCounts::<T>::get();
```

Comment:
So we sent the number of *processed* - not received - messages?  
What do we do with the error count? Maybe we should also ACK those.